rescan.go

```diff
@@ -593,20 +591,30 @@ rescanLoop:
 						curStamp.Height, curStamp.Hash)
 
 					current = false
-					s.unsubscribeBlockMsgs(subscription)
-					subscription = nil
+					blockSubscription.Cancel()
+					blockSubscription = nil
 				}
 
-			case header := <-blockConnected:
-				err := handleBlockConnected(header)
-				if err != nil {
-					log.Errorf("Unable to process new "+
-						"block %v", header.BlockHash(),
-						err)
-				}
+			case ntfn := <-blockSubscription.Notifications:
```

Comment:
👍 will likely also end up resolving some issues in the integration tests caused by a non-strict ordering of connected vs disconnected notifications